CIL wrongly carrying __attribute__((__no_sanitize__("...")))

Source file `nosanitize.c`

```c
#include <stdint.h>

__attribute__((__no_sanitize__("signed-integer-overflow"))) static uint32_t add(uint32_t a, int b) {
    return a + b;
}

int main() {
    uint32_t x = 1;
    int y = 2;
    if (add(x, y) < 1) {
        return 1;
    }
    return 0;
}
```

CIL output `nosanitize.cil.c`

```c
/* ...... */
int main(void) /* comes from pVDecl with beginsFunDef; vinline is really false and the varinfo, magic 53242549633104, also has 0 entries in vvardecls */ 
{ 
  uint32_t x ;
  int y ;
  uint32_t __attribute__((__no_sanitize__("signed-integer-overflow")))  tmp ;

  {
#line 8
  x = (uint32_t )1;
#line 9
  y = 2;
#line 10
  tmp = add(x, y);
#line 10
  if (tmp < (uint32_t __attribute__((__no_sanitize__("signed-integer-overflow")))  )1) {
#line 11
    return (1);
  }
#line 13
  return (0);
}
}
```

Clang-20 wrapped in allocscc output (`CC_IS_CLANG=1 CC=clang-20 allocscc -c nosanitize.c`)

```
nosanitize.c: In function 'add':
nosanitize.c:4:3: warning: 'no_sanitize' attribute does not apply to types [-Wattributes]
    4 |     return a + b;
      |   ^ ~~~~
nosanitize.c: In function 'main':
nosanitize.c:11:3: warning: 'no_sanitize' attribute ignored [-Wattributes]
   11 |         return 1;
      |   ^     ~~
nosanitize.c:10:3: warning: 'no_sanitize' attribute does not apply to types [-Wattributes]
   10 |     if (add(x, y) < 1) {
      |   ^ 
```

The attribute string with `no_sanitize` should not be carried along the `uint32_t` type to create temporaries and casts in the example above.

--- a/src/cil.ml
+++ b/src/cil.ml
@@ -1479,7 +1479,7 @@ let attributeHash: (string, attributeClass) H.t =
   List.iter (fun a -> H.add table a (AttrFunType false))
     [ "format"; "regparm"; "longcall"; 
       "noinline"; "always_inline"; "gnu_inline"; "leaf"; "cold"; "alloc_size";
-      "artificial"; "warn_unused_result"; "nonnull"; "pure";
+      "artificial"; "warn_unused_result"; "nonnull"; "pure"; "no_sanitize";
     ];
 
   List.iter (fun a -> H.add table a (AttrFunType true))

